raccoongang/frontend-app-course-authoring
# feat: Course unit - Rendering xblocks in iframes

### Dependencies
[Edx-platform](https://github.com/raccoongang/edx-platform/tree/ts-develop) ts-develop branch

### Description

This PR demonstrates the xblock content display functionality for the course unit page. The current implementation is a demonstration. Work on refactoring and displaying the Drag and Drop block continues.

The primary features were implemented:
- Rendering xblocks inside iframes (Advanced, Discussion, Text, Open Response, Problem, Video).
- The work of internal xblock scripts has been partially implemented.

### Developer notes
- The source for this functionality is [frontend-app-library-authoring](https://github.com/openedx/frontend-app-library-authoring).
- Technically, our goal is to visually display the content of the xblock + configure the work of the internal scripts of the xblock that are rendered inside the iframe (if it does not require a large investment of time).

### Testing instructions
1. Run master devstack.
2. Start platform make dev.up.lms+cms+frontend-app-course-authoring and make checkout on this branch.
3. Enable the new Unit page by adding a waffle flag `contentstore.new_studio_mfe.use_new_unit_page` in the CMS admin panel.
4. Make sure that the MFE setting `ENABLE_UNIT_PAGE=true` is enabled.
5. Сomment out [the addition of the Ajax prefix](https://github.com/openedx/edx-platform/blob/master/cms/static/cms/js/main.js#L17) in the edx-platform (temporary solution).
6. Go to the Course Unit page from the Course Outline page.

https://github.com/raccoongang/frontend-app-course-authoring/assets/93188219/f90b8a5d-6e97-4c27-bc6a-021db6596b3d
